morpheus/modules/file_to_df.py

```diff
@@ -0,0 +1,262 @@
+
+    schema_config = config.get("schema", None)
+    schema_str = schema_config.get("schema_str", None)
+    encoding = schema_config.get("encoding", None)
+    file_type = config.get("file_type", None)
+    filter_null = config.get("filter_null", None)
+    parser_kwargs = config.get("parser_kwargs", None)
+    cache_dir = config.get("cache_dir", None)
+
+    download_method: typing.Literal["single_thread", "multiprocess", "dask",
+                                    "dask_thread"] = os.environ.get("MORPHEUS_FILE_DOWNLOAD_TYPE", "multiprocess")
+    cache_dir = os.path.join(cache_dir, "file_cache")
+
+    # Load input schema
+    schema = pickle.loads(bytes(schema_str, encoding))
+
+    try:
+        file_type = str_to_file_type(file_type.lower())
+    except Exception:
+        raise ValueError("Invalid input file type '{}'. Available file types are: CSV, JSON".format(file_type))
+
+    def get_dask_cluster():
+
+        try:
+            import dask
+            from dask.distributed import LocalCluster
+        except ModuleNotFoundError:
+            raise Exception("Install 'dask' and 'distributed' to allow file downloads using dask mode.")
+
+        logger.debug("Creating dask cluster...")
+
+        # Up the heartbeat interval which can get violated with long download times
+        dask.config.set({"distributed.client.heartbeat": "30s"})
+
+        dask_cluster = LocalCluster(start=True, processes=not download_method == "dask_thread")
+
+        logger.debug("Creating dask cluster... Done. Dashboard: %s", dask_cluster.dashboard_link)
+
+        return dask_cluster
+
+    def get_dask_client(dask_cluster):
+
+        from dask.distributed import Client
+
+        logger.debug("Creating dask client...")
+        dask_client = Client(dask_cluster)
+        logger.debug("Creating dask client %s ... Done.", dask_client)
+
+        return dask_client
+
+    def close_dask_cluster():
+        if (dask_cluster is not None):
+            logger.debug("Stopping dask cluster...")
+            dask_cluster.close()
+
+            logger.debug("Stopping dask cluster... Done.")
+
+    def single_object_to_dataframe(file_object: fsspec.core.OpenFile,
+                                   file_type: FileTypes,
+                                   filter_null: bool,
+                                   parser_kwargs: dict):
+
+        retries = 0
+        s3_df = None
+        while (retries < 2):
+            try:
+                with file_object as f:
+                    s3_df = read_file_to_df(f,
+                                            file_type,
+                                            filter_nulls=filter_null,
+                                            df_type="pandas",
+                                            parser_kwargs=parser_kwargs)
+
+                break
+            except Exception as e:
+                if (retries < 2):
+                    logger.warning("Refreshing S3 credentials")
+                    # cred_refresh()
```

Comment:
Commented code. Should this be removed or re-enabled?

Reply:
this can be removed.